### PR TITLE
Feature: Easter Egg Hunt Waypoints

### PR DESCRIPTION
**Pull request by RayDeeUx**
## What
Easter Egg hunt waypoints for 2024.

<details>
<summary>Images</summary>

image

</details>

## Changelog New Features
+ Easter Egg Hunt 2024 waypoints. - Erymanthus + walker

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/event/lobby/waypoints/easter/EggWaypoints.kt [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    fun onChat(event: LorenzChatEvent) {
+        if (!config.allWaypoints && !config.allEntranceWaypoints) return
+        if (!isEgg) return
+
+        if (!HypixelData.hypixelLive) return
+        if (LorenzUtils.inSkyBlock) return
+
+        val message = event.message
+        if (message.startsWith("§a§lYou found an Easter Egg! §r") || message == "§aYou have received the §bsuper reward§a!" || message == "§cYou already found this egg!") {
+            val Egg = Egg.entries.minByOrNull { it.waypoint.distanceSqToPlayer() }!!
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/event/lobby/waypoints/easter/EggWaypoints.kt]: Local variable name 'Egg' should start with a lowercase letter

File: src/main/java/at/hannibal2/skyhanni/features/event/lobby/waypoints/easter/EggEntrances.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
+package at.hannibal2.skyhanni.features.event.lobby.waypoints.easter
+
+import at.hannibal2.skyhanni.utils.LorenzVec
+
+enum class EggEntrances(
+    val eggEntranceName: String,
+    val waypoint: LorenzVec,
+    vararg val egg: Egg,
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/event/lobby/waypoints/easter/EggEntrances.kt]: this variable needs a better name

File: src/main/java/at/hannibal2/skyhanni/features/event/lobby/waypoints/easter/EggEntrances.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package at.hannibal2.skyhanni.features.event.lobby.waypoints.easter
+
+import at.hannibal2.skyhanni.utils.LorenzVec
+
+enum class EggEntrances(
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/event/lobby/waypoints/easter/EggEntrances.kt:5]: prob rename to LobbyEasterEggEntrance

File: src/main/java/at/hannibal2/skyhanni/features/event/lobby/waypoints/easter/Egg.kt [hunk 4]
```diff
@@ -0,0 +1,39 @@
+package at.hannibal2.skyhanni.features.event.lobby.waypoints.easter
+
+import at.hannibal2.skyhanni.utils.LorenzVec
+
+enum class Egg(val eggName: String, val waypoint: LorenzVec) {
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/event/lobby/waypoints/easter/Egg.kt]: prob rename to LobbyEasterEgg

File: src/main/java/at/hannibal2/skyhanni/features/event/lobby/waypoints/easter/EggWaypoints.kt [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    @SubscribeEvent
+    fun onRenderWorld(event: LorenzRenderWorldEvent) {
+        if (!HypixelData.hypixelLive) return // don't show outside live hypixel network (it's disabled on alpha)
+        if (LorenzUtils.inSkyBlock) return
+        if (!isEgg) return
+
+        if (config.allWaypoints) {
+            for (Egg in Egg.entries) {
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/event/lobby/waypoints/easter/EggWaypoints.kt]: Local variable name 'Egg' should start with a lowercase letter

**Review by CalMWolfs — CHANGES_REQUESTED** [review]

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Cal's reviews

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
other changes look good just a bit more cleanup, see #1207

**Pull request merged**